### PR TITLE
Update dev docs

### PR DESCRIPTION
**Pull request by mythmon**

### CODE REVIEW

File: docs/dev/concepts.rst [hunk 1]
```diff
@@ -1,55 +1,37 @@
...
-
-     // Actions should return a Promise when executed
-     async execute() {
-       this.normandy.log(this.recipe.arguments.message, 'info');
-     }
-   }
-
-   // registerAction is in the sandbox's global object
-   registerAction('console-log', ConsoleLogAction);
+For example, a survey action would include code to check whether we have
```
Comment (glasserc) [inline docs/dev/concepts.rst]: This paragraph is quite disjoint with the previous one which is about action configuration. I think you're trying to summarize the whole subsection. In that case, maybe something like: `For example, a "survey" action consists of code within Firefox that [...]. When issued to clients, this action is paired with configuration that [...].`

File: docs/dev/concepts.rst [hunk 2]
```diff
@@ -60,93 +42,74 @@ perform. Recipes contain three important pieces of information:
...
+Normandy Client
+---------------
+The Normandy client is `a Firefox component`_ that reads and executes
+instructions from the server. Upon activation (typically every few hours),
+the runtime:
 
 1. Downloads :ref:`recipes <recipes>` from the Normandy service.
 2. Verifies the signature of the recipes.
 3. Evaluates the recipe filters and filters out recipes that do not match the
    client the runtime is installed within.
```
Comment (glasserc) [inline docs/dev/concepts.rst]: I know this isn't exactly part of the change but I think `the runtime is installed within` is a little clunky. Maybe just end the sentence after `the client`.

File: docs/dev/concepts.rst [hunk 3]
```diff
@@ -60,93 +42,74 @@ perform. Recipes contain three important pieces of information:
...
-4. Downloads the :ref:`actions <actions>` for the remaining recipes.
-5. Executes the action code for each recipe in a sandbox, passing in the
-   arguments from the recipe, and a :doc:`driver object </dev/driver>` containing
-   methods that can perform privileged actions.
-
-The runtime is implemented as `a component of Firefox in mozilla-central`_.
+4. Passes the recipes to the corresponding action code for execution.
```
Comment (glasserc) [inline docs/dev/concepts.rst]: This sentence is strongly situated within the point of view of the code itself. How about `Executes the corresponding action code using the recipe's arguments`? Details can be found in mozilla-central anyhow..

File: docs/dev/concepts.rst [hunk 4]
```diff
@@ -60,93 +42,74 @@ perform. Recipes contain three important pieces of information:
...
 ^^^^^^^^^^^^^^^^^^^^^^^^^
-Actions and recipes that are downloaded by the runtime are signed according to
-the Content-Signature protocol as provided by the autograph_ service. The
-runtime verifies the signature upon downloading the recipes and actions,
-ensuring that the runtime only executes recipes that have been signed with a
-Mozilla-controlled key.
+Recipes that are downloaded by the runtime are signed according to the
+Content-Signature protocol as provided by the Autograph_ service. The client
+verifies the signature upon downloading the recipes, ensuring that the
+runtime only executes recipes that have been signed with a Mozilla-controlled
```
Comment (glasserc) [inline docs/dev/concepts.rst]: It feels like you're trying to remove the word `runtime` here so maybe replace it with `client`. Also, you might want to change `are signed` to `are each signed`.

File: docs/dev/concepts.rst [hunk 5]
```diff
@@ -158,16 +121,13 @@ distributed by the service.
...
-
-To perform actions that JavaScript normally can't (such as displaying a
-Heartbeat survey), the action in the sandbox is passed a
-:doc:`driver object </dev/driver>`, which contains methods that can modify the
-client or trigger other privileged behavior.
+Static Instructions
+^^^^^^^^^^^^^^^^^^^
+Recipe data is provided in either a static format (JSON) or an executable
+format designed specifically to be weak and without access to exfiltrate
+data(Jexl filters). This means that even if an attacker could run a malicious
```
Comment (glasserc) [inline docs/dev/concepts.rst]: Space after `data`.

File: docs/dev/concepts.rst [hunk 6]
```diff
@@ -158,16 +121,13 @@ distributed by the service.
...
-limits the access of the JavaScript to prevent it from modifying Firefox in ways
-that haven't been reviewed and approved beforehand.
-
-To perform actions that JavaScript normally can't (such as displaying a
-Heartbeat survey), the action in the sandbox is passed a
-:doc:`driver object </dev/driver>`, which contains methods that can modify the
-client or trigger other privileged behavior.
+Static Instructions
+^^^^^^^^^^^^^^^^^^^
+Recipe data is provided in either a static format (JSON) or an executable
```
Comment (glasserc) [inline docs/dev/concepts.rst]: Here is the second place where you used `static` that made me think something was amiss. I guess you are using the word `static` here to indicate that the instructions are not in the form of JavaScript code. I am not really sure this is an important distinction (I have a friend who likes to point out that not only code is data, but also data is code) but if so you might simply say that recipe information is not Turing-complete.

**Comment by glasserc** [comment]
The docs failed to build but I'm not really sure why. I'm going to rerun the job.

**Comment by glasserc** [comment]
OK, this time it failed with an error:

```
Warning, treated as error:
/home/circleci/repo/docs/dev/concepts.rst:69:Unknown directive type "info".

.. info::
```

File: docs/dev/concepts.rst [hunk 7]
```diff
@@ -1,55 +1,37 @@
...
+perform a variety of actions to aid the user and gather feedback to guide
+Mozilla in developing a better browser for them. To do this, we need a way to
+instruct Firefox to perform certain actions without having to download an
+update or wait for a new release.
 
-SHIELD is split into three main concepts: Actions, Recipes, and the runtime.
-This document describes those concepts in detail.
+The system is made of two main components, a server and client. These
+components are known simply as the Normandy client and Normandy server. The
+server provides static information that the client acts on.
```
[the remark this answers is not in the document]
Comment (mythmon) [reply docs/dev/concepts.rst]: What I was trying to convey was more leaning towards your second option, that the data is non-executable. I'll tweak the notes to try and convey both things.

**Comment by mythmon** [comment]
bors r= @glasserc

**Pull request merged**